soda_machine.rb

```diff
@@ -7,12 +7,16 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.size
   end
 
   def find_soda(soda_brand)
+    @sodas.find {|soda| soda.brand == soda_brand}
   end
 
   def sell(soda_brand)
```

Comment:
Good. I'd prefer not running `find_soda` twice. I'd argue a local variable to hold the result for the duration of a method execution is preferable to the processing (although minimal) of running the `find_soda` method twice.